nautobot_ssot/integrations/dna_center/utils/nautobot.py

```diff
@@ -86,3 +97,58 @@ def assign_version_to_device(adapter, device: Device, software_lcm: UUID):
         destination_id=device.id,
     )
     new_assoc.validated_save()
+
+
+if METADATA_FOUND:
+
+    def object_has_metadata(obj: object) -> Optional[bool]:
+        """Check if object has MetadataType assigned to it.
+
+        Args:
+            obj (object): Object to check for MetadataType.
+
+        Returns:
+            bool: True if MetadataType is found, False otherwise.
+        """
+        try:
+            ObjectMetadata.objects.get(
+                assigned_object_id=obj.id,
+                metadata_type=MetadataType.objects.get(name="Last Sync from DNA Center"),
+            )
+            return True
+        except (MetadataType.DoesNotExist, ObjectMetadata.DoesNotExist):
+            return False
+
+    def add_or_update_metadata_on_object(adapter: Adapter, obj: object, scoped_fields: List[str]) -> ObjectMetadata:  # pylint: disable=inconsistent-return-statements
+        """Add or Update Metadata on passed object and assign scoped fields.
+
+        Args:
+            adapter (Adapter): Adapter that has logging facility.
+            obj (object): Object to assign Metadata to.
+            scoped_fields (List[str]): List of strings that are scoped fields for object.
+
+        Returns:
+            ObjectMetadata: Metadata object that is created or updated.
+        """
+        if METADATA_FOUND:
```

Comment:
Can we turn this one around so it is a guard clause, like

```
if not METADATA_FOUND:
    continue
```

I prefer it this way, because its less indentation context to keep in mind when reading the code.

Reply:
Done.